tests/build_/test_build_requestv2.py

```diff
@@ -0,0 +1,690 @@
+        kwargs['isolated_build_node_selector'] = built_type_nodeselectors['isolated']
+        br.set_params(**kwargs)
+        build_json = br.render(api=osbs)
+
+        if expected:
+            assert build_json['spec']['nodeSelector'] == expected
+        else:
+            assert 'nodeSelector' not in build_json['spec']
+
+    @pytest.mark.parametrize(('reactor_config_map', 'valid'), [  # noqa:F811
+        # No api so fail
+        (None, False),
+        # api does not have the config-map, fail
+        ('reactor-config-map', False),
+        # api has a valid config map
+        ('reactor-config-map', True),
+    ])
+    def test_set_config_map(self, osbs, reactor_config_map, valid):  # noqa:F811
+        with open(os.path.join(INPUTS_PATH, "config_map.json")) as fp:
+            raw = fp.read()
+        mock = flexmock(sys.modules['__builtin__' if six.PY2 else 'builtins'])
+        mock.should_call('open')  # set the fall-through
+        (mock.should_receive('open')
+            .with_args('inputs/config_map.json')
+            .and_return(flexmock(read=lambda: raw)))
+
+        build_request = BuildRequestV2(INPUTS_PATH)
+        kwargs = get_sample_prod_params()
+        kwargs['reactor_config_map'] = reactor_config_map
+        build_request.set_params(**kwargs)
+        if reactor_config_map:
+            config_map_data = {
+                'required_secrets': ['Nothing', 'Nada'],
+                'worker_token_secrets': ['Secret']
+            }
+            config_map = osbs.create_config_map(reactor_config_map, config_map_data)
+            if valid:
+                flexmock(osbs).should_receive('get_config_map').and_return(config_map)
+            else:
+                flexmock(osbs).should_receive('get_config_map').and_return(None)
+            test_api = osbs
+        else:
+            test_api = None
+
+        if valid:
+            build_json = build_request.render(api=test_api)
+            json_env = build_json['spec']['strategy']['customStrategy']['env']
+            envs = {}
+            for env in json_env:
+                envs[env['name']] = env.get('valueFrom', None)
+
+                configmapkeyref = {
+                    'name': reactor_config_map,
+                    'key': 'config.yaml'
+                }
+
+            assert 'REACTOR_CONFIG' in envs
+            assert 'configMapKeyRef' in envs['REACTOR_CONFIG']
+            assert envs['REACTOR_CONFIG']['configMapKeyRef'] == configmapkeyref
+            custom_secrets = build_json['spec']['strategy']['customStrategy']['secrets']
+            all_secrets = set(['Nothing', 'Nada', 'Secret'])  # use py2 syntax for compatibility
+            configured_secrets = set(secret['secretSource']['name'] for secret in custom_secrets)
+            assert all_secrets == configured_secrets
+            for secret in custom_secrets:
+                assert secret['secretSource']['name'] in all_secrets
```

Comment:
These last two lines are redundant because we know that `all_secrets == configured_secrets`.

Reply:
Fixed.